### PR TITLE
Introduce Fedora and Firefox CaC profiles for common workstation users

### PR DESCRIPTION
**Pull request by j-ode**
#### Description:

- In my bachelor's thesis, I created a new security policy for common Fedora workstation users. To implement this policy, I introduce two new Common User Security Profiles: cusp_fedora and cusp_firefox. During the implementation of the policy in CaC, many issues were encountered. Most notably, the cusp_firefox profile is a temporary measure til an issue with CaC buildsystem #10462 about inclusion of multiple products in one profile is resolved. The goal is to only have cusp_fedora in the future that would incorporate all cusp_firefox rules.
- I also introduce two new CaC rules, firefox_policy-content_blocker and package_gnome_software_installed, both of which are templated. The former ensures that a content blocker called uBlock Origin is installed in Firefox, while the latter ensures that the GNOME Software package is installed in Fedora.
- The rest of the changes are just fixing prodtypes and platforms to make rules applicable to Fedora, as well as fixing issues with a Firefox OVAL template.
- Even though many issues discovered during the implementation remain unresolved, the profiles work in the current state, and the aim is to ideally get them into the June CaC release, so that it could be presented during my thesis defense.

#### Rationale:

- Currently, the Fedora profiles in CaC are largely unmaintained and not useful for common workstation users. The RHEL8/9 profiles and the standards they are based on are not practical for common users due to a variety of issues. The new profile is based on a security policy that is based on the CIS RHEL 8 - workstation level 1 benchmark, and incorporates rules from ANSSI and STIG as well. Unlike those standards however, the focus was on balancing usability with security with common users of Fedora workstation in mind.
- The inclusion of a maintained Fedora profile in CaC would also open the door for activation of OAA in Fedora, which could lead to A) larger adoption of this profile and of CaC/OpenSCAP as a whole, and B) would enable us to test OAA in Fedora as well as in RHEL.

### CODE REVIEW

**Review by Mab879 — DISMISSED** [review]
Thanks for the PR! I think will be good addition to the project. I have a couple of nitpicks on top of the CI findings.

**Review by ggbecker — DISMISSED** [review]
Two gating items I can already see that will need to be addressed,

- the order in which the `prodtype` `fedora` has been put. It needs to be in the alphabetical order
- I noticed at least one file without the new line at then end `controls/cusp_firefox.yml`, but there is more. If you used vscode, I suggest enabling the option to add newline at the end whenever a file is saved. You can press `Ctrl+,` (Control + Comma), it will open the preferences window, then type `Insert Final Newline` and make sure it is ticked.

**Comment by ggbecker** [comment]
> Two gating items I can already see that will need to be addressed,
> 
>     * the order in which the `prodtype` `fedora` has been put. It needs to be in the alphabetical order
> 
>     * I noticed at least one file without the new line at then end `controls/cusp_firefox.yml`, but there is more. If you used vscode, I suggest enabling the option to add newline at the end whenever a file is saved. You can press `Ctrl+,` (Control + Comma), it will open the preferences window, then type `Insert Final Newline` and make sure it is ticked.

@j-ode I remember we have this: https://github.com/ComplianceAsCode/content/blob/master/utils/autoprodtyper.py please use it.

File: controls/cusp_fedora.yml [hunk 1]
```diff
@@ -0,0 +1,517 @@
...
+  - id: 5.1
+    title: General network configuration
+    description: >-
+      If the user did not configure IPv6 on the system and it is not needed, it should be disabled.
+    status: partial
+    rules:
+      - kernel_module_sctp_disabled
+      - kernel_module_dccp_disabled
+      - sysctl_net_ipv4_ip_forward
+      - sysctl_net_ipv6_conf_all_forwarding
```
Comment (yuumasato) [inline controls/cusp_fedora.yml:434]: Is the use of VMs a supported use case for the profiles? 

Once IP forwarding is disabled in a host the guest VMs will loose access to the external world.

Comment (j-ode) [reply controls/cusp_fedora.yml:434]: Thanks for pointing this out, I suppose this profile should be VM-friendly as well. I looked at the two rules, and the ipv4 one has a jinja macro in the platform: https://github.com/ComplianceAsCode/content/blob/f31f13744364d29489e2a414a6a3fd43957ca3bd/shared/macros/01-general.jinja#LL972C1-L977C17 that disables the rule if rhel8 product uses oVirt. I could create a new macro that would apply the machine platform for fedora product only, but I wonder if that could have adverse effect on other potential fedora profiles that do not consider the VM usecase. What are your thoughts on such a solution @yuumasato?

**Comment by Mab879** [comment]
> > Two gating items I can already see that will need to be addressed,
> > ```
> > * the order in which the `prodtype` `fedora` has been put. It needs to be in the alphabetical order
> > 
> > * I noticed at least one file without the new line at then end `controls/cusp_firefox.yml`, but there is more. If you used vscode, I suggest enabling the option to add newline at the end whenever a file is saved. You can press `Ctrl+,` (Control + Comma), it will open the preferences window, then type `Insert Final Newline` and make sure it is ticked.
> > ```
> 
> @j-ode I remember we have this: https://github.com/ComplianceAsCode/content/blob/master/utils/autoprodtyper.py please use it.

This case `./utils/fix_rules.py sort_prodtype` might be more useful.

File: controls/cusp_fedora.yml (hunk 1 above)
Comment (Mab879) [reply controls/cusp_fedora.yml:434]: You may be able to use CPE applicability language here. Something like `package[libvirtd]` as a platform on Fedora might do it.

File: controls/cusp_fedora.yml [hunk 2]
```diff
@@ -0,0 +1,517 @@
...
+    title: Security policy selection
+    description: >-
+      Users should apply the “Fedora Common User Security Policy” in the installer.
+    status: manual
+
+  - id: 2.2
+    title: Disk partitioning
+    description: >-
+      Users should put the /home, /tmp, /var, /var/tmp and /var/log directories on separate partitions.
+    status: manual
```
Comment (Mab879) [inline controls/cusp_fedora.yml:42]: The rules from `linux_os/guide/system/software/disk_partitioning` might be useful here

**Review by Mab879 — COMMENTED** [review]
Please ensure the all descriptions are in sentence case (first letter of sentence is capitalized). The title keys should be title case.

Also, my original comments from my first review still stand, it seems that one of the rebases musts have gotten rid of your changes.

File: controls/cusp_fedora.yml (hunk 2 above)
Comment (j-ode) [reply controls/cusp_fedora.yml:42]: Yes, the issue is that the respective Anaconda remediations block the installation when the disk is not correctly partitioned which is not ideal, I talk about this in the thesis and created an issue in OAA https://github.com/OpenSCAP/oscap-anaconda-addon/issues/236

File: controls/cusp_fedora.yml (hunk 1 above)
Comment (yuumasato) [reply controls/cusp_fedora.yml]: I guess making the rules for IP forwarding not applicable when an VM hypervisor is installed makes sense.

But could there be usecases that expect VMs to not have access to the external world?

**Comment by j-ode** [comment]
The Automatus CS8/CS9 CI fails are expected, as package_gnome_software_installed is a fedora-only rule and therefore cannot be present in the RHEL8/9 benchmarks as the errors suggest

File: controls/cusp_fedora.yml (hunk 1 above)
Comment (j-ode) [reply controls/cusp_fedora.yml]: The problem is that libvirt is installed by default on Fedora 38 Workstation:
image
The purpose of this profile is to increase security but at the same time maintain usability, so I believe that users should be able to run VMs on their machines without connectivity problems (most users probably want networking inside their VMs). I am not sure there is a way to include these rules in the profile.

**Review by jan-cerny — APPROVED** [review]
The fail in centos8 testing farm job is related to rule rsyslog_files_permissions and will be fixed by https://github.com/ComplianceAsCode/content/pull/10540, therefore it isn't related to the contents of this PR.

**Pull request merged**